lib/vector-buffers/src/buffer_usage_data.rs

```diff
@@ -95,16 +100,25 @@ pub struct BufferUsageData {
     received_byte_size: AtomicU64,
     sent_event_count: AtomicU64,
     sent_byte_size: AtomicU64,
-    dropped_event_count: Option<AtomicU64>,
+    dropped_event_data: Option<BufferUsageDroppedEventData>,
     max_size_bytes: AtomicU64,
     max_size_events: AtomicUsize,
 }
 
+#[derive(Debug)]
+struct BufferUsageDroppedEventData {
+    dropped_event_count: AtomicU64,
+    dropped_event_size: AtomicU64,
+}
+
 impl BufferUsageData {
     pub fn new(mode: WhenFull, idx: usize) -> Self {
-        let dropped_event_count = match mode {
+        let dropped_event_data = match mode {
             WhenFull::Block | WhenFull::Overflow => None,
-            WhenFull::DropNewest => Some(AtomicU64::new(0)),
+            WhenFull::DropNewest => Some(BufferUsageDroppedEventData {
+                dropped_event_count: AtomicU64::new(0),
+                dropped_event_size: AtomicU64::new(0),
+            }),
```

Comment:
I think this could be simplified by adding a `#[derive(Default)]` to `BufferUsageDroppedEventData` and then:
```suggestion
            WhenFull::DropNewest => Some(Default::default()),
```